overlord/devicestate/devicemgr.go

```diff
@@ -987,11 +1011,6 @@ func checkGadgetOrKernel(st *state.State, snapInfo, curInfo *snap.Info, flags sn
 		return nil
 	}
 
-	if release.OnClassic {
```

Comment:
@niemeyer I cannot parse this, atm this branch is still blocking installing kernel on classic, some lines up. Do you want to unblock that now? it seems confusing, it would just put some blobs on disk and we need to stop some copying actions